components/eamxx/src/physics/rrtmgp/eamxx_rrtmgp_process_interface.cpp

```diff
@@ -382,6 +402,48 @@ void RRTMGPRadiation::initialize_impl(const RunType /* run_type */) {
   // Determine whether or not we are using a fixed solar zenith angle (positive value)
   m_fixed_solar_zenith_angle = m_params.get<double>("Fixed Solar Zenith Angle", -9999);
 
+  // Not sure what to do about this
+  constexpr int ps = SCREAM_PACK_SIZE;
+
+  // Get time-varying input fields and their associated file(s)
+  // TODO: Harden this impl; get the EKAT msg/warn stuff and use it here to avoid bad logic at the outset
+  // TODO: Maybe also move this to the very top?
+  m_time_varying_active_gases_list =
+      m_params.get<std::vector<std::string>>("time_varying_active_gases_list", {"nolistgiven"});
+  m_time_varying_active_gases_file =
+      m_params.get<std::vector<std::string>>("time_varying_active_gases_file", {"nofilegiven"});
+
+  // Initialize the time interpolator
+  if (m_time_varying_active_gases_file[0] != "nofilegiven") {
+    m_time_interp =
+        util::TimeInterpolation(m_grid, m_time_varying_active_gases_file);
+
+    // To create helper fields for to interpolate their values
+    for (std::string name : m_time_varying_active_gases_list) {
+      // Helper fields that will temporarily store the input values
+      std::string name_transient = name + "_transient";
+      Field field_transient;
+      if (name == "co2" || name == "n2o" || name == "ch4" || name == "f11" || name == "f12") {
+        // These are all scalar (0d) time-varying
+        FieldLayout scalar0d_layout{{}, {}};
```

Comment:
This is subtle. The brace initialization causes some things to trigger. In this case
```c++
FieldLayout layout0d { {}, {} };
```
the compiler interprets the input as an initializer list of size 2. It then scans the constructors of `FieldLayout` and finds `FieldLayout (const std::initializer_list<FieldTag>& tags);`, so it converts the input to an initializer list of tags, of size 2. In this case, otoh,
```c++
FieldLayout layout1d { {LEV}, {nlevs} };
```
the compiler cannot parse the input as a single initializer list, since the two entries would have different type. Scanning the class, it finds
```c++
  FieldLayout (const std::vector<FieldTag>& tags, const std::vector<int>& dims);
```
and since it can construct the two vectors from the two inner init lists, it realizes it can call this constructor. Notice that, if you do
```c++
FieldLayout layout0d ( {}, {} );
```
the compiler can no longer interpret this as a single init list with two entries, and is forced to pick the constructor that takes two vectors (like for layout1d).